cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/ACLStoreTableDataset.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, eitherap express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.api.dataset.lib.AbstractDataset;
+import co.cask.cdap.api.dataset.lib.IndexedObjectStore;
+import co.cask.cdap.api.dataset.module.EmbeddedDataset;
+import co.cask.common.authorization.ACLEntry;
+import co.cask.common.authorization.Permission;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Simple dataset implementation of {@link co.cask.common.authorization.ACLStore}.
+ *
+ * TODO: Optimize
+ */
+public class ACLStoreTableDataset extends AbstractDataset implements ACLStoreTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ACLStoreTableDataset.class);
+  private static final Gson GSON = new Gson();
+
+  private final IndexedObjectStore<ACLEntry> store;
+
+  public ACLStoreTableDataset(DatasetSpecification spec, @EmbeddedDataset("acls") IndexedObjectStore<ACLEntry> store) {
+    super(spec.getName(), store);
+    this.store = store;
+  }
+
+  @Override
+  public void write(ACLEntry entry) throws Exception {
+    byte[][] secondaryKeys = getSecondaryKeysForWrite(entry);
+    store.write(getKey(entry), entry, secondaryKeys);
+  }
+
+  @Override
+  public boolean exists(ACLEntry entry) throws Exception {
+    if (entry.getPermission() == Permission.ANY) {
+      List<ACLEntry> any = store.readAllByIndex(getKey(new ACLEntry(entry.getObject(), entry.getSubject(), null)));
```

Comment:
I think so. Can you give some more details?

Anyway, I think if we also index ACLs by Permission.ANY in `getSecondaryKeysForWrite`, then there wouldn't need to be any `Permission.ANY` handling in here.